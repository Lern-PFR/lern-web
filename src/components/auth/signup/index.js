import SignUpForm from './SignUpForm';

export {
	// eslint-disable-next-line import/prefer-default-export
	SignUpForm,
};
