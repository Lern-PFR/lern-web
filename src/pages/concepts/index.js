import ConceptDetailsPage from './ConceptDetailsPage';

export {
	// eslint-disable-next-line import/prefer-default-export
	ConceptDetailsPage,
};
