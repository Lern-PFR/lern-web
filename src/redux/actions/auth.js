export const loginAction = ({ user }) => ({
    type: 'AUTH_LOGIN',
    payload: { user}
});

export const logoutAction = () => ({
    type: 'AUTH_LOGOUT'
});