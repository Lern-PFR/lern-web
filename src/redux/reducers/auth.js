import localstorage from '../../lib/localstorage';

const initialState = {
    currentUser: null,
};

export default (state = initialState, { type, payload }) => {
    switch (type) {
        case 'AUTH_LOGIN': {
            const newState = {
                ...state,
                currentUser: payload.user,
            };
            localstorage.put('user', JSON.stringify(payload.user));

            return newState;
        }
        case 'AUTH_LOGOUT': {
            localStorage.removeItem('user');
            return {
                ...initialState,
            };
        }
    }
}