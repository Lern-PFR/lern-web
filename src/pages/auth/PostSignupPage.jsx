import PropTypes from 'prop-types';
import { withTranslation } from 'react-i18next';
import { StyledDiv, StyledSvg } from 'components/shared/styledElements';
import { BodyCopy, Canon } from 'components/shared/typography';
import conf from 'conf';
import { hero, layout, subtitle } from 'theme/pages/auth/postSignupPage';
import { illustrationSvg } from 'theme/pages/auth/signUpPage';

/**
 * @name PostSignupPage
 * @description The page towards which the user is redirected after completing the signup process.
 *
 * @author TimothÃ©e Simon-Franza
 *
 * @param {func} t	The translation method provided by the withTranslation HoC.
 */
const PostSignupPage = ({ t }) => (
	<StyledDiv {...layout}>
		<StyledDiv>
			<Canon as="h1" {...hero}>{t('authentication.pages.post_signup.hero')}</Canon>
			<BodyCopy {...subtitle}>{t('authentication.pages.post_signup.subtitle')}</BodyCopy>
		</StyledDiv>
		<StyledSvg src={`${conf.svgIllustrationsPath}/mail_box.svg`} {...illustrationSvg} />
	</StyledDiv>
);

PostSignupPage.propTypes = {
	t: PropTypes.func.isRequired,
};

export default withTranslation()(PostSignupPage);
