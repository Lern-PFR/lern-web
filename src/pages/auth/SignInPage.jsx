import { useCallback } from 'react';
import { useDispatch } from 'react-redux';
import PropTypes from 'prop-types';
import { withTranslation } from 'react-i18next';
import { SignInForm } from 'components/auth/signin';
import { StyledDiv, StyledSvg } from 'components/shared/styledElements';
import { BodyCopy, Canon } from 'components/shared/typography';
import { login } from 'redux/actions/users';
import { hero, illustrationSvg, layout, subtitle } from 'theme/pages/auth/signInPage';
import conf from 'conf';
import routes from 'routes';

/**
 * @name SignInPage
 * @description A page accessible to unauthenticated users to sign into the application.
 *
 * @author TimothÃ©e Simon-Franza
 *
 * @param {func} t	The translation method provided by the withTranslation HoC.
 */
const SignInPage = ({ t }) => {
	const dispatch = useDispatch();

	const onSubmit = useCallback((formData) => {
		const credentials = {
			username: formData?.username,
			password: formData?.password,
		};

		dispatch(login(credentials, routes.home.default));
	}, [dispatch]);

	return (
		<StyledDiv {...layout}>
			<StyledDiv>
				<Canon {...hero} tag="h1">{t('authentication.pages.signin.hero')}</Canon>
				<BodyCopy {...subtitle}>{t('authentication.pages.signin.subtitle')}</BodyCopy>
				<SignInForm onSubmit={onSubmit} />
			</StyledDiv>
			<StyledSvg src={`${conf.svgIllustrationsPath}/sign_in.svg`} {...illustrationSvg} />
		</StyledDiv>
	);
};

SignInPage.propTypes = {
	t: PropTypes.func.isRequired,
};

export default withTranslation()(SignInPage);
