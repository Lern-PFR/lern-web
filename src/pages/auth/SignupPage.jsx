import SignUpForm from 'components/auth/signup/SignUpForm';
import { StyledDiv, StyledSvg } from 'components/shared/styledElements';
import { Canon, DoublePica } from 'components/shared/typography';
import PropTypes from 'prop-types';
import { useCallback } from 'react';
import { withTranslation } from 'react-i18next';
import { connect } from 'react-redux';
import { compose } from 'redux';
import { signUp } from 'redux/actions/users';
import { hero, illustrationSvg, layout, subtitle } from 'theme/pages/auth/signUpPage';
import conf from 'conf';

/**
 * @name SignUpPage
 * @description A page accessible to unauthenticated users to create an account.
 *
 * @author TimothÃ©e Simon-Franza
 *
 * @param {func} 	dispatchSignUp	Dispatched action creator used to send the user creation request to the API.
 * @param {func}	t				The translation method provided by the withTranslation HoC.
 */
const SignUpPage = ({ dispatchSignUp, t }) => {
	const onSubmit = useCallback((formData) => {
		console.log(formData);

		if (false) {
			dispatchSignUp(formData);
		}
	}, [dispatchSignUp]);

	return (
		<StyledDiv {...layout}>
			<StyledDiv>
				<Canon {...hero} tag="h1">{t('authentication.pages.signup.hero')}</Canon>
				<DoublePica {...subtitle}>{t('authentication.pages.signup.subtitle')}</DoublePica>
				<SignUpForm onSubmit={onSubmit} />
			</StyledDiv>
			<StyledSvg src={`${conf.svgIllustrationsPath}/sign_up.svg`} {...illustrationSvg} />
		</StyledDiv>
	);
};

SignUpPage.propTypes = {
	dispatchSignUp: PropTypes.func.isRequired,
	t: PropTypes.func.isRequired,
};

/**
 *
 * @param {*} state
 */
const mapStateToProps = (state) => {
	const { users: { currentUser } } = state;

	return {
		currentUser,
	};
};

const mapDispatchToProps = {
	dispatchSignUp: signUp,
};

export default compose(
	withTranslation(),
	connect(mapStateToProps, mapDispatchToProps)
)(SignUpPage);
