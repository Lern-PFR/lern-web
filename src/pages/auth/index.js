import SignupPage from './SignupPage';
import PostSignupPage from './PostSignupPage';

export {
	SignupPage,
	PostSignupPage,
};
