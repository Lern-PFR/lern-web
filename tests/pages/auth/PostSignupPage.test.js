import { shallow } from 'enzyme';
import { PostSignupPage } from 'pages/auth';

describe('Post Sign-up page', () => {
	it('should match previous snapshot.', () => {
		const wrapper = shallow(<PostSignupPage />);

		expect(wrapper).toMatchSnapshot();
	});
});
