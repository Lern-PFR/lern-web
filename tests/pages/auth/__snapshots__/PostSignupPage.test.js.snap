// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post Sign-up page should match previous snapshot. 1`] = `
<styled.div
  alignItems="center"
  display="grid"
  gridGap="3em"
  gridTemplateColumns="minmax(400px, 1fr) 1fr"
  height="100%"
  marginX="auto"
  maxWidth="1450px"
>
  <styled.div>
    <Canon
      as="h1"
      lineHeight="initial"
    >
      authentication.pages.post_signup.hero
    </Canon>
    <BodyCopy
      lineHeight="initial"
    >
      authentication.pages.post_signup.subtitle
    </BodyCopy>
  </styled.div>
  <Styled(ReactSVG)
    display="inline-block"
    src="/assets/svg/illustrations/mail_box.svg"
  />
</styled.div>
`;
